src/Form/tests/FormControl.test.jsx

```diff
@@ -31,4 +49,21 @@ describe('FormControl', () => {
     expect(onChangeFunc).toHaveBeenCalledTimes(inputText.length);
     expect(ref.current.style.height).toEqual(`${ref.current.scrollHeight + ref.current.offsetHeight}px`);
   });
+
+  it('should apply and accept input mask for phone numbers', () => {
+    render(<Component />);
+
+    act(() => {
+      const input = screen.getByTestId('form-control-with-mask');
+      userEvent.type(input, '1234567890');
```

Comment:
[curious] generally, `userEvent.*` functions are called outside of `act` since the `userEvent.*` functions are already wrapped with `act` ([example source](https://github.com/testing-library/user-event/issues/497#issuecomment-729904582)).

Reply:
@adamstankiewicz thank you for bringing this issue to our attention 👍
At the moment we have a [lot of similar examples](https://github.com/search?q=repo%3Aopenedx%2Fparagon%20act(&type=code)) of wrapping in `act` `userEvent` in Paragon. I've looked into several interesting sources (including the [one](https://github.com/testing-library/user-event/issues/497#issuecomment-729904582) you shared), this is really an unnecessary solution, here are some confirmations:
1. [When should I use act() in react-testing-library?](https://flyingsquirrel.medium.com/when-should-i-use-act-in-react-testing-library-d7dd22a3340e)
2. [Internal implementation](https://github.com/testing-library/react-testing-library/blob/main/src/pure.js#L21-L23)
3. [Is wrapping with act now required?](https://github.com/testing-library/user-event/discussions/906#discussioncomment-2522468)

My solution is based on [Is wrapping with act now required?](https://github.com/testing-library/user-event/discussions/906#discussioncomment-2522468). As it turned out, we have differences in the `@testing-library/dom` subdependency in the versions of the `@testing-library/react@12.1.5` and `@testing-library/user-event@13.5.0` libraries. A dependency conflict negatively affects the performance of the `userEvent` and we see a warning after running tests.

<img width="611" alt="image" src="https://github.com/openedx/paragon/assets/93188219/b1bc125a-14d4-4e04-9488-b115734c550a">

**We have several options for further action:**
1. Apply now the solution with an override of the `@testing-library/dom` package (with version `9.3.3`). This way we will get rid of the problem of displaying warnings in the console and our tests will be cleaner. This solution can be developed in a separate task, where we remove the wrapping in `act` `userEvent`. I have verified this in the Tabs component and in my current case (FormControl).
2. Make these changes and “synchronize” the `@testing-library/dom` subdependency in `@testing-library/react` and `@testing-library/user-event` in the pull request for [updating to React 18](https://github.com/openedx/paragon/pull/2774). So It is quite likely that we will not have to do this since we will update the versions of `@testing-library/react` and `@testing-library/user-event`. But we cannot predict whether `@testing-library/dom` versions will change in the future. Therefore, the override option seems workable.

An [article about dependency overriding](https://github.com/npm/rfcs/blob/main/accepted/0036-overrides.md#motivation), in which we can make sure that our intentions are correct:

> There should only be a single copy of a given dependency in our tree, but npm's dependency resolution will result in a duplicate due to version conflicts.

Reply:
Thanks for digging into, this @PKulkoRaccoonGang! TIL that `@testing-library/dom` has to be synchronized :) I agree with your synopsis/options; interesting that I'm not sure this has come up before in MFEs that use RTL. But I agree the `overrides` is sufficient to hold over until the React 18 upgrade.